keras/src/backend/torch/export.py

```diff
@@ -1,35 +1,152 @@
+        # Disable false alarms related to lifting parameters.
+        warnings.filterwarnings("ignore", message=".*created when tracing.*")
+        warnings.filterwarnings(
+            "ignore", message=".*Unable to find the path of the module.*"
+        )
 
-        if isinstance(resource, layers.Layer):
-            # Variables in the lists below are actually part of the trackables
-            # that get saved, because the lists are created in __init__.
-            variables = resource.variables
-            trainable_variables = resource.trainable_variables
-            non_trainable_variables = resource.non_trainable_variables
-            self._tf_trackable.variables += tree.map_structure(
-                self._convert_to_tf_variable, variables
+        if not isinstance(resource, torch.nn.Module):
+            raise TypeError(
+                "`resource` must be an instance of `torch.nn.Module`. "
+                f"Received: resource={resource} (of type {type(resource)})"
             )
-            self._tf_trackable.trainable_variables += tree.map_structure(
-                self._convert_to_tf_variable, trainable_variables
+
+        def _check_input_signature(input_spec):
+            for s in tree.flatten(input_spec.shape):
+                if s is None:
+                    raise ValueError(
+                        "The shape in the `input_spec` must be fully "
+                        f"specified. Received: input_spec={input_spec}"
+                    )
+
+        def _to_torch_tensor(x, replace_none_number=1):
+            shape = backend.standardize_shape(x.shape)
+            shape = tuple(
+                s if s is not None else replace_none_number for s in shape
             )
-            self._tf_trackable.non_trainable_variables += tree.map_structure(
-                self._convert_to_tf_variable, non_trainable_variables
+            return ops.ones(shape, x.dtype)
+
+        tree.map_structure(_check_input_signature, input_signature)
+        sample_inputs = tree.map_structure(_to_torch_tensor, input_signature)
+        sample_inputs = tuple(sample_inputs)
+
+        # Ref: torch_xla.tf_saved_model_integration
+        # TODO: Utilize `dynamic_shapes`
+        exported = torch.export.export(
+            resource, sample_inputs, dynamic_shapes=None, strict=False
+        )
+        options = torch_xla.stablehlo.StableHLOExportOptions(
+            override_tracing_arguments=sample_inputs
+        )
+        stablehlo_model = torch_xla.stablehlo.exported_program_to_stablehlo(
+            exported, options
+        )
+        state_dict_keys = list(stablehlo_model._bundle.state_dict.keys())
+
+        # Remove unused variables.
+        for k in state_dict_keys:
+            if "lifted" not in k:
+                stablehlo_model._bundle.state_dict.pop(k)
+
+        def _mangle_tf_root_scope_name(name):
+            # TF has more restricted constrain on the variable names.
```

Comment:
Please add a comment to explain the transformation being done (regex aren't super readable)

Reply:
This function was copied from torch_xla which should be unnecessary for the case. Removed.